src/Normalizer/FileEntityNormalizer.php

```diff
@@ -92,7 +92,7 @@ public function denormalize($data, $class, $format = NULL, array $context = [])
     $file_data = (string) $this->httpClient->get($data['uri'][0]['value'])->getBody();
 
     $path = 'temporary://' . $this->fileSystem->basename($data['uri'][0]['value']);
-    $data['uri'] = file_unmanaged_save_data($file_data, $path);
+    $data['uri'] = FileSystemInterface::saveData($file_data, $path);
```

Comment:
So this actually blows up with: `Non-static method Drupal\Core\File\FileSystemInterface::saveData() cannot be called statically`

Looking at the [change record](https://www.drupal.org/node/3006851) probably should inject the `file_system` service if we're wanting to be proper.

Reply:
yup - that's what i was thinking